### PR TITLE
Select information Front-End (No CSS)

### PR DESCRIPTION
**Pull request by SivanC**
Implements the logic for the Select Information page, allowing the user to select which information of theirs they would like to include in their generated code. App.js is temporary for testing since I did not have access to the header and footer etc. Submission button will need to be hooked up to the Generate QR Code page when it's done. Will follow up with CSS in another PR.

Related to user story #10 and task #21

### CODE REVIEW

File: front-end/src/SelectInformationForm.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    console.log('Submitted!')
+  }
+
+  // get data from api
+  useEffect(() => {
+    const fetchData = async () => {
+      console.log('Fetching user information data...')
+      try {
+        const response = await axios(
+          'https://my.api.mockaroo.com/info.json?key=820f1130')
```
Comment (byroncjy) [inline front-end/src/SelectInformationForm.js:34]: Might be good to change the mock data to have mock values of Platform and Information instead so it's clearer what this page does if there's a demo. No big deal though

File: front-end/src/SelectInformationForm.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        console.log('Unable to retrieve data!')
+        console.error(err)
+      }
+    }
+    fetchData()
+  }, [])
+
+  // initialize checked after data is set
+  useEffect(() => {
+    setChecked(Array.from({ length: data.length }, () => true))
```
Comment (byroncjy) [inline front-end/src/SelectInformationForm.js:47]: From my understanding, this sets all fields to true (checked) upon loading? Should we instead save the user's choices of whether a field has been checked or not from last QR code generation and set those checked choices upon loading? 
API schema will probably need to change though

**Review by byroncjy — COMMENTED** [review]
Looks good, just a few things with the logic and display I thought we should consider

We also need to resolve merge conflicts

File: front-end/src/SelectInformationForm.js (hunk 1 above)
Comment (SivanC) [reply front-end/src/SelectInformationForm.js:34]: Yeah agreed, I didn't see a good option but if I have time this week I might change the API to have custom data for that purpose.

File: front-end/src/SelectInformationForm.js (hunk 2 above)
Comment (SivanC) [reply front-end/src/SelectInformationForm.js:47]: Your understanding is correct; I think either choice is equally okay, ideally in the future we could add custom presets so you could just press one button. I think in general that people are not going to have more than like ten fields, so it should be fine for now, let me know if you disagree.

Comment (byroncjy) [reply front-end/src/SelectInformationForm.js:47]: Custom presets is also another option I hadn't thought of. yeah ideally either one would work

**Review by byroncjy — APPROVED** [review]

**Pull request merged**